crates/server/src/verifywalletsignatures/near.rs

```diff
@@ -52,6 +95,66 @@ pub fn verify_near_signature(
     verify(public_key_str, &payload_hash, signature_base64).is_ok()
 }
 
+pub async fn check_for_near_account_key(
+    current_near_root_key: &str,
+    account_name: &str,
+    rpc_url: &str,
+) -> EyreResult<bool, ApiError> {
+    let client = Client::new();
+    // Check if root key belongs to the account
+    let body = json!({
+        "jsonrpc": "2.0",
+        "id": "dontcare",
+        "method": "query",
+        "params": {
+            "request_type": "view_access_key",
+            "finality": "final",
+            "account_id": account_name,
+            "public_key": current_near_root_key,
+        }
+    });
```

Comment:
Out of interest, it might be nice to be formalising places like this into structs...